src/app/features/file/couchdb-file.service.ts

```diff
@@ -49,24 +57,54 @@ export class CouchdbFileService extends FileService {
     super(entityMapper, entities, logger);
   }
 
-  uploadFile(file: File, entity: Entity, property: string): Observable<any> {
+  uploadFile(
+    file: File,
+    entity: Entity,
+    property: string,
+    compression?: number
+  ): Observable<any> {
     const obs = this.requestQueue.add(
-      this.runFileUpload(file, entity, property)
+      this.runFileUpload(file, entity, property, compression)
     );
     this.reportProgress($localize`Uploading "${file.name}"`, obs);
+    this.cache[`${entity.getId(true)}/${property}`] = obs.pipe(
+      last(),
+      map((blob: Blob) => URL.createObjectURL(blob)),
+      shareReplay()
+    );
     return obs;
   }
 
-  private runFileUpload(file: File, entity: Entity, property: string) {
-    const blob = new Blob([file]);
-    const attachmentPath = `${this.attachmentsUrl}/${entity.getId(true)}`;
-    return this.getAttachmentsDocument(attachmentPath).pipe(
-      concatMap(({ _rev }) =>
-        this.http.put(`${attachmentPath}/${property}?rev=${_rev}`, blob, {
-          headers: { "Content-Type": file.type, "ngsw-bypass": "" },
-          reportProgress: true,
-          observe: "events",
-        })
+  private runFileUpload(
+    file: File,
+    entity: Entity,
+    property: string,
+    compression: number
+  ) {
+    let blobObs: Observable<Blob>;
+    if (compression) {
```

Comment:
what happens here if the file is not an image?

For a clean "separation of concerns" we may want to keep resizing out of the general file service and just call that separately from the photo component?

Reply:
Good point, that actually made things muuuuch simpler 😅